payments-core/src/main/java/com/stripe/android/payments/core/injection/AuthenticationComponent.kt

```diff
@@ -0,0 +1,18 @@
+package com.stripe.android.payments.core.injection
+
+import com.stripe.android.payments.core.authentication.DefaultIntentAuthenticatorRegistry
+import dagger.Component
+import javax.inject.Singleton
+
+/**
+ * [Component] for com.stripe.android.payments.core.authentication.
+ *
+ * It holds the dagger graph for [DefaultIntentAuthenticatorRegistry], with
+ * more dependencies daggerized and a higher level [Component]s created, this class will be merged
+ * into it.
+ */
+@Singleton
```

Comment:
not a strong opinion, just state the problems we ran into [0] and suggest considering creating custom scopes. 
In Terminal we always use @Singleton above their constructor and assume it's singleton globally. Assuming so, we don't even add them into a dagger module [example](https://git.corp.stripe.com/stripe-internal/android/blob/d156ad47/core/modules/paymentcollection/src/main/kotlin/com/stripe/core/paymentcollection/PaymentCollectionCoordinator.kt#L42). and now that we have multiple components, it's hard to tell which component it lives in.
where as if you demand a custom scope (e.g. @AuthComponentScope), and always make modules:
```
@Module
interface AuthModule() {
    @Provides
    @AuthScope 
     fun providesAuthRegistry(...) : AuthRegistry
}

@AuthScope // redundant annotation but making it clear.
class AuthRegistry(...)
```
 it's very explicit in their mindset that its lifecycle is bound to `AuthComponent`. And if say `LoginComponent` wants to use `AuthRegistry`, people can easily see it's in a different scope, therefore a different component should provide it instead of putting `AuthRegistry` in their own dagger graph. But i don't know how hard it is to change all existing scopes. 

And if you use subcomponent, it MIGHT also lead to conflicts [1].

[0] https://paper.dropbox.com/doc/Reader-app-creates-two-instances-of-Singleton-classes--BLnJswz18bTpDhhhBiHqqBTRAg-MFHcHBstiz1U8cGnRMP6o

[1] No subcomponent may be associated with the same scope as any ancestor component, although two subcomponents that are not mutually reachable can be associated with the same scope because there is no ambiguity about where to store the scoped objects. (The two subcomponents effectively have different scope instances even if they use the same scope annotation.) https://dagger.dev/dev-guide/subcomponents.html

Reply:
Thanks for the heads up, created a dedicated @AuthenticationScope  , we might face this issue when we have a dedicated dagger component in modules introduced later.

Reply:
reverted back to @Singleton after offline sync.
The issue on Terminal is that the app and the SDK each has a @Component that provides an instance, but for our case, the client app never interacts with any internal objects with the SDK and will not create any @Component, so as long as our SDK maintains a single @Component to cover the entire dagger graph, we can make sure the `@Singleton` s are indeed singleton